heidegger_index/validators.py

```diff
@@ -0,0 +1,47 @@
+from django.core.exceptions import ValidationError
+
+import re
+
+## https://wiki.dnb.de/pages/viewpage.action?pageId=48139522
+
+def validate_gnd(value):
+    # ONLY WORKS FOR PND / POST-2012 entries.
+
+    # Validate type
+    if type(value) == str or type(value) == int:
+        # Validate with regex according to https://www.wikidata.org/wiki/Property:P227
+        GND_REGEX = r"1[012]?\d{7}[0-9X]|[47]\d{6}-\d|[1-9]\d{0,7}-[0-9X]|3\d{7}[0-9X]"
+        regex = re.compile(GND_REGEX, re.IGNORECASE)
+        if not regex.match(value):
+            raise ValidationError(
+                f"{value} is not syntactically valid."
+                )
+
+        if type(value) == str:
+            value = re.sub(r"[^0-9X]", "", value)
+
+        sum = 0
+        for i, d in enumerate(reversed(value)):
+            if i == 0:
+                control_digit = d
+                continue
+            sum = sum + (int(d) * (i + 1))
+
+        print(sum)
```

Comment:
Haal deze print statements er even tussenuit, of vervang het door een enkele, human readable regel als je graag feedback zou willen zien.